File: python/2.2.2_Thermistor.py
#!/usr/bin/env python3

import PCF8591 as ADC
import RPi.GPIO as GPIO
import time
import math

GPIO.setmode(GPIO.BCM)

def print_message():
	print ("========================================")
	print ("|               Thermistor             |")
	print ("|    ------------------------------    |")
	print ("|           SDA,SCL connect to         |")
	print ("|   the corresponding pin of PCF8591   |")
	print ("|                                      |")	
	print ("| Use thermistor to detect temperature |")
	print ("|                                      |")
	print ("|                            SunFounder|")
	print ("========================================\n")
	print ('Program is running...')
	print ('Please press Ctrl+C to end the program...')
	input ("Press Enter to begin\n")

def setup():
	ADC.setup(0x48)

def loop():
	while True:
		analogVal = ADC.read(0)
		Vr = 5 * float(analogVal) / 255
		Rt = 10000 * Vr / (5 - Vr)
		temp = 1/(((math.log(Rt / 10000)) / 3950) + (1 / (273.15+25)))
		temp = temp - 273.15
		print ('temperature = ', temp, 'C')
		time.sleep(0.2)

if __name__ == '__main__':
	print_message()
	try:
		setup()
		loop()
	except KeyboardInterrupt: 
		pass	
		
	

File: c/2.2.2/2.2.2_Thermistor.c
/**********************************************************************
* Filename    : 2.2.2_Thermometer.c
* Description : Read temperature
* Author      : Robot
* E-mail      : support@sunfounder.com
* website     : www.sunfounder.com
* Update      : Cavon    2016/07/01
**********************************************************************/
#include <stdio.h>
#include <wiringPi.h>
#include <pcf8591.h>
#include <math.h>

#define		PCF     120

int main()
{
	unsigned char analogVal;
	double Vr, Rt, temp;
	printf("\n");
	printf("\n");
	printf("========================================\n");
	printf("|               Thermistor             |\n");
	printf("|    ------------------------------    |\n");
	printf("|           SDA,SCL connect to         |\n");
	printf("|   the corresponding pin of PCF8591   |\n");
	printf("|                                      |\n");	
	printf("| Use thermistor to detect temperature |\n");
	printf("|                                      |\n");
	printf("|                            SunFounder|\n");
	printf("========================================\n");
	printf("\n");
	printf("\n");
	if(wiringPiSetup() == -1){
		printf("setup wiringPi failed !");
		return 1;
	}
	// Setup pcf8591 on base pin 120, and address 0x48
	pcf8591Setup(PCF, 0x48);

	while(1) // loop forever
	{
		printf("loop");
		analogVal = analogRead(PCF + 0);
		Vr = 5 * (double)(analogVal) / 255;
		Rt = 10000 * (double)(Vr) / (5 - (double)(Vr));
		temp = 1 / (((log(Rt/10000)) / 3950)+(1 / (273.15 + 25)));
		temp = temp - 273.15;
		printf("Current temperature : %lf\n", temp);

		delay (200);
	}
	return 0;
}